### ISSUE TITLE
Game does not scale to window shape

### ISSUE THREAD

**Issue by TheArcaneBrony** [jdah/minecraft-weekend#17]
Game does not account for different window shapes, causing everyting (especially the UI) to be squished on other aspect ratios, such as 4:3.
![image](https://user-images.githubusercontent.com/13570458/107879977-4c7c3100-6edc-11eb-9d7c-570515d02125.png)

**Comment by bayi** [comment]
Add this to src/gfx/window.c:

Somewhere near the top insert this lines:
```
static void _refresh_callback(GLFWwindow* window) {
    printf("Window refresh\n");
    glfwSwapBuffers(window);
}
```

Then after line 104 add this:
```
    glfwSetWindowRefreshCallback(window.handle, _refresh_callback);
```

This solves the resize event handling in glfw

**Comment by bayi** [comment]
Nevermind this only fixes the blank window issue with compositing window managers, the ui scaling issue still applies ...

**Comment by jdah** [comment]
Fixed with 01efb83539f7d12a859264d213f02ce8c8693bb2.
